sdk/src/types/block/output/delegation.rs

```diff
@@ -187,9 +187,17 @@ impl DelegationOutputBuilder {
             unlock_conditions,
         };
 
-        output.amount = match self.amount {
-            OutputBuilderAmount::Amount(amount) => amount,
-            OutputBuilderAmount::MinimumAmount(params) => output.minimum_amount(params),
+        match self.amount {
+            OutputBuilderAmount::Amount(amount) => output.amount = amount,
```

Comment:
Should `delegated_amount` be removed from the new method and be set here as well?
Because `delegated_amount` gets set in the `new()` method of the builder and amount can be set to a different value in `with_amount()` so they can end up with different values

Reply:
tbh I don't know. I think it depends a lot on whether the amounts should always match

Reply:
The transaction is invalid if it doesn't match

Reply:
So I think we shouldn't allow to set the delegated amount, first time it should be set to the amount and delayed claiming it must stay the same as for creation

Reply:
doesn't that mean it's always the same as the amount?

Reply:
For delayed claiming, the amount can be reduced, but the delegated amount field must still stay as before

Reply:
True

Reply:
Sorry, will you revert or just change it a bit different?